src/ui/EditableTask.ts

```diff
@@ -16,4 +25,209 @@ export type EditableTask = {
     forwardOnly: boolean;
     blockedBy: Task[];
     blocking: Task[];
-};
+
+    constructor(editableTask: {
+        // NEW_TASK_FIELD_EDIT_REQUIRED
+        description: string;
+        status: Status;
+        priority: EditableTaskPriority;
+        recurrenceRule: string;
+        createdDate: string;
+        startDate: string;
+        scheduledDate: string;
+        dueDate: string;
+        doneDate: string;
+        cancelledDate: string;
+        forwardOnly: boolean;
+        blockedBy: Task[];
+        blocking: Task[];
+    }) {
+        this.description = editableTask.description;
+        this.status = editableTask.status;
+        this.priority = editableTask.priority;
+        this.recurrenceRule = editableTask.recurrenceRule;
+        this.createdDate = editableTask.createdDate;
+        this.startDate = editableTask.startDate;
+        this.scheduledDate = editableTask.scheduledDate;
+        this.dueDate = editableTask.dueDate;
+        this.doneDate = editableTask.doneDate;
+        this.cancelledDate = editableTask.cancelledDate;
+        this.forwardOnly = editableTask.forwardOnly;
+        this.blockedBy = editableTask.blockedBy;
+        this.blocking = editableTask.blocking;
+    }
+
+    public static fromTask(task: Task, allTasks: Task[]) {
```

Comment:
I presumed that the return-type of this would be an `EditableTask` and tried writing a minimal `EditableTask.test.ts` test for it - to demonstrate round-tripping `Task` -> `EditableTask` - `Task`.

I learned that the return types of both `fromTask()` and `applyEdits()` were a bit non-obvious...

I don't know how far you're going to be taking this, but it looks to me like it it's good enough that it could exist for some time.

In which case, some of the following would be helpful:

- Get WebStorm to add explicit return types
- Add JSdocs, including linking to related methods
- A minimal Jest test that shows the round-trip process...

Reply:
Thanks! I tried doing that and discovered some untended behaviour. Also is it hard to test all the fields of a `Task` to show the diff. I think we need some kind of helper for that. However all of this is a concern for later and another PR, so I tried to snapshot the current behaviour as is.

Reply:
> Thanks! I tried doing that and discovered some untended behaviour. Also is it hard to test all the fields of a `Task` to show the diff. I think we need some kind of helper for that. However all of this is a concern for later and another PR, so I tried to snapshot the current behaviour as is.

For info (not suggesting you use it now) but `toMatchTaskDetails()` is pretty good for this. It's powerful magic, but nicely expressive.